### PR TITLE
added tests to check filedialog options on different OSes (#344)

### LINKED ISSUE #344: Should have a unit-test that asserts the filedialog options for the diff platforms when choosing a deployment
follows up on #318, #337, #341 

Given that we now had bugs for all 3 platforms, we should assert that the dialog options are set up correctly for the different platforms via a unit-test.

### PR DESCRIPTION
**Pull request by lstocchi**
it fixes #344

### CODE REVIEW

**Comment by adietish** [comment]
@lstocchi dare to rebase this please?

File: test/serverExplorer.test.ts [hunk 1]
```diff
@@ -314,4 +314,117 @@ suite('Server explorer', () => {
...
+        });
+
+        test('check promise get rejected if dialog is closed without choosing', async () => {
+            sandbox.stub(serverExplorer, 'quickPickDeploymentType' as any).resolves(undefined);
+
+            try {
+                await serverExplorer.addDeployment(undefined);
+                expect.fail();
+            } catch (err) {
+                expect(err).equals(undefined);
```
Comment (adietish) [inline test/serverExplorer.test.ts]: just making sure we got this right: even though the promise is rejected, there's no error being displayed in vscode if we a user cancels the file chooser (esc or cancelling the dialog), correct?

Comment (lstocchi) [reply test/serverExplorer.test.ts]: No there is no error displayed if a user cancel the dialog or click the esc button. It's their choice to esc/terminate it. In this solution there is still the old bug solved in PR #348 . Don't know if you refer to this.

Comment (lstocchi) [reply test/serverExplorer.test.ts]: > just making sure we got this right: even though the promise is rejected, there's no error being displayed in vscode if we a user cancels the file chooser (esc or cancelling the dialog), correct?

Added new unit test for previous PR #348 and improved existing ones. Check if it makes sense for you now.

**Pull request merged**